pkg/cluster/serviceBinding.go

```diff
@@ -145,24 +149,16 @@ func performBinding(ctx context.Context, options *ServiceBindingOptions, ns stri
 		Spec: v1alpha1.ServiceBindingSpec{
 			BindAsFiles: true,
 			Services:    []v1alpha1.Service{serviceRef},
-			Application: &appRef,
+			Application: appRef,
 		},
 	}
 	sb.SetGroupVersionKind(v1alpha1.GroupVersionKind)
 
-	if options.ForceUseSDK {
-		return useSDKForBinding(clients, sb)
-	}
-
 	// Check of operator is installed
 	_, err := clients.dynamicClient.Resource(v1alpha1.GroupVersionResource).Namespace(ns).
 		List(ctx, metav1.ListOptions{Limit: 1})
 	if err != nil {
-		if options.ForceUseOperator {
-			return localizer.MustLocalizeError("cluster.serviceBinding.operatorMissing", localize.NewEntry("Error", err))
-		}
-		logger.Debug("Service binding Operator not available. Will use SDK option for binding")
-		return useSDKForBinding(clients, sb)
+		return localizer.MustLocalizeError("cluster.serviceBinding.operatorMissing", localize.NewEntry("Error", err))
```

Comment:
we discussed that yesteday that we going to map errors specifically for providing more reliable info. 
So we should probably:

- Log error that we recognize 
- Return error to be printed again with more generic/non i18n content

Reply:
Yes - in the commands it makes sense to map errors and preserving the error object does not matter too much. However in an SDK scenario (like this one) I think it is important to preserve it because the place it is being used is able act different based on the error type.

Reply:
In this case error will be cryptic - 404 kafka API doesn't exist or missing auth token. Do you suggest to return original error only?

Reply:
No it can be wrapped like this:

```go
return fmt.Errorf("your reason: %w", err)
```

`%w` means wrap the error.